mlflow/store/artifact/s3_artifact_repo.py

```diff
@@ -33,6 +33,7 @@ def _cached_get_s3_client(
     access_key_id=None,
     secret_access_key=None,
     session_token=None,
+    region_name=None,
```

Comment:
Are these needed?
OptimizedS3ArtifactRepository needs this definition, which is a subclass of CloudArtifactRepository, which in turn is a subclass of ArtifactRepository. 
Does the S3ArtifactRepository have some use for `region_name`?

Reply:
It does not have any use. The `OptimizedS3ArtifactRepository` needs to set the `region_name` for the S3 client and imports `_get_s3_client` function from here the `S3ArtifactRepository`. The reason I didn't duplicate the `_get_s3_client` function because I didn't think it would be necessary for them to have separate caches.